### PR TITLE
Revert: Update prepare_package.dart to use the process_runner package instead having duplicate code. #62710

### LINKED ISSUE #63615: `prepare_package.dart` crashing with `NoSuchMethodError: The method 'start' was called on null`
This crashed on the [dev packaging builder](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8872148853048735808/+/steps/prepare__create_and_publish_a_flutter_archive/0/stdout), caused by https://github.com/flutter/flutter/pull/62710. Note that each packaging build uses the latest version of this script.

It was fixed by https://github.com/flutter/flutter/pull/63608.

### PR DESCRIPTION
**Pull request by gspencergoog**
This reverts #62710, since it has a `noSuchMethodError` in some circumstances.

This isn't just a straight `git revert`, because I had to resolve the conflicts in the reverted `pubspec.yaml` files and re-run `flutter update-packages --force-upgrade` to make the checksums match.

Fixes https://github.com/flutter/flutter/issues/63615

### CODE REVIEW

**Comment by christopherfujino** [comment]
@jonahwilliams can you take a look at the pubspec changes, to make sure they're right?

File: packages/flutter_tools/pubspec.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ dependencies:
   mustache_template: 1.0.0+1
   package_config: 1.9.3
   process: 3.0.13
-  fake_async: 1.1.0-nullsafety
+  quiver: 2.1.3
```
Comment (jonahwilliams) [inline packages/flutter_tools/pubspec.yaml:28]: This is wrong and needs to be reverted

File: dev/integration_tests/flutter_gallery/pubspec.yaml [hunk 2]
```diff
@@ -51,7 +51,6 @@ dev_dependencies:
   flutter_goldens:
     sdk: flutter
   test: 1.16.0-nullsafety.1
-  e2e: 0.7.0
```
Comment (jonahwilliams) [inline dev/integration_tests/flutter_gallery/pubspec.yaml:54]: this seems suspicious too

File: packages/flutter_tools/pubspec.yaml (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter_tools/pubspec.yaml]: Reverted

File: dev/integration_tests/flutter_gallery/pubspec.yaml (hunk 2 above)
Comment (gspencergoog) [reply dev/integration_tests/flutter_gallery/pubspec.yaml]: Also reverted.

**Comment by gspencergoog** [comment]
I reverted those spurious changes, and re-ran the update to get a new checksum.  Seems to be OK now. PTAL.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**